HW3/P3/sum.cl

```diff
@@ -48,8 +63,16 @@ __kernel void sum_blocked(__global float* x,
     // 
     // Be careful that each thread stays in bounds, both relative to
     // size of x (i.e., N), and the range it's assigned to sum.
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE
+    lSize = get_local_size(0);
+    gID = get_global_id(0);
+    if (k-1 < N - k*gID){
+        minS = k;
+    }
+    else{
+        minS = N - k*gID;
+    }
```

Comment:
You can simplify the above if statement, and just put in the following for loop condition:
...; j < k*(gID +1) && (k*gID + j) < N; ...